server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java

```diff
@@ -304,6 +312,34 @@ private static QueryBuilder toValueQuery(MetricCriterion criterion) {
     }
   }
 
+  public List<String> searchTags(@Nullable String textQuery, int pageSize) {
```

Comment:
it's the same for rules and issues tags. You cannot have pagination for an ES facet. That's what we're using here to return tags. Still, we want to be able to change the number of tags returned.